homeassistant/components/google/__init__.py

```diff
@@ -213,24 +232,26 @@ def setup(hass, config):
     if not os.path.isfile(token_file):
         do_authentication(hass, config, conf)
     else:
-        if not check_correct_scopes(token_file):
+        if not check_correct_scopes(token_file, conf):
             do_authentication(hass, config, conf)
         else:
             do_setup(hass, config, conf)
 
     return True
 
 
-def check_correct_scopes(token_file):
+def check_correct_scopes(token_file, config):
     """Check for the correct scopes in file."""
     with open(token_file) as tokenfile:
-        if "readonly" in tokenfile.read():
+        contents = tokenfile.read()
+        target_scope = f"\"scope\": \"{config.get(CONF_CALENDAR_ACCESS).scope}\""
```

Comment:
I know I asked for this to be more robust -- but maybe just check the scope to avoid over asserting on the json format.  I am only saying this because I saw https://github.com/home-assistant/core/pull/53364 a few days ago where folks were compressing json and then it had a subtle side effect of breaking a regexp.

Reply:
Reduced down to quoted scope. This is needed because `read_write` scope is a subset of `read_only` scope.

Reply:
I completely missed that, oof, so I get it why the original check was just for the suffix then.  Are you still happy with this approach?

Reply:
The new approach is fine. I think you were right to bring up the potential risk of whitespace messing with the check, and this new check is a bit more robust.